apps/prairielearn/src/tests/questionSharing.test.ts

```diff
@@ -0,0 +1,252 @@
+import ERR = require('async-stacktrace');
+import { assert } from 'chai';
+import { step } from 'mocha-steps';
+import { config } from '../lib/config';
+import fetch from 'node-fetch';
+import helperClient = require('./helperClient');
+import helperServer = require('./helperServer');
+import sqldb = require('@prairielearn/postgres');
+const sql = sqldb.loadSqlEquiv(__filename);
+import { EXAMPLE_COURSE_PATH, TEST_COURSE_PATH } from '../lib/paths';
+
+import syncFromDisk = require('../sync/syncFromDisk');
+
+import { makeMockLogger } from './mockLogger';
+const { logger } = makeMockLogger();
+
+const siteUrl = 'http://localhost:' + config.serverPort;
+const baseUrl = siteUrl + '/pl';
+
+const UUID_REGEXP = /[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}/;
+
+const testCourseId = 1;
+const testCourseSharingName = 'test-course';
+const exampleCourseId = 2;
+const exampleCourseSharingName = 'example-course';
+const sharingSetName = 'share-set-example';
+
+function sharingPageUrl(courseId) {
+  return `${baseUrl}/course/${courseId}/course_admin/sharing`;
+}
+
+async function setSharingName(courseId, name) {
+  const sharingUrl = sharingPageUrl(courseId);
+  const response = await helperClient.fetchCheerio(sharingUrl);
+
+  const token = response.$('#test_csrf_token').text();
+  return await fetch(sharingUrl, {
+    method: 'POST',
+    headers: { 'Content-type': 'application/x-www-form-urlencoded' },
+    body: new URLSearchParams({
+      __action: 'unsafe_choose_sharing_name',
+      __csrf_token: token,
+      course_sharing_name: name,
+    }).toString(),
+  });
+}
+
+async function accessSharedQuestionAssessment() {
+  const assessmentsUrl = `${baseUrl}/course_instance/${exampleCourseId}/instructor/instance_admin/assessments`;
+  const assessmentsPage = await helperClient.fetchCheerio(assessmentsUrl);
+  const sharedQuestionAssessmentUrl =
+    siteUrl +
+    assessmentsPage
+      .$(`a:contains("Example of Importing Questions From Another Course")`)
+      .attr('href');
+  const res = await helperClient.fetchCheerio(sharedQuestionAssessmentUrl);
+  assert.equal(res.ok, true);
+  return res;
+}
+
+describe('Question Sharing', function () {
+  this.timeout(80000);
+
+  describe('Create a sharing set and add a question to it', () => {
+    let exampleCourseSharingId;
+
+    step('set up testing server', helperServer.before(TEST_COURSE_PATH));
+    after('shut down testing server', helperServer.after);
+
+    step('ensure course has question sharing enabled', async () => {
+      await sqldb.queryAsync(sql.enable_question_sharing, {});
+    });
+
+    step('Sync coures with sharing enabled', (callback) => {
+      syncFromDisk.syncOrCreateDiskToSql(EXAMPLE_COURSE_PATH, logger, function (err, result) {
```

Comment:
There's an async variant of this function; let's use that instead!

Reply:
done